### PR TITLE
Add a reader for Meris L2 data

### PR DESCRIPTION
**Pull request by johannesocean**
New reader for ENVISAT-MERIS data. The 4th generation reprocessing of MERIS data implements a Sentinel 3 like format including Level 2 products and TOA-radiances in the same dataset package. Hence, we can use the same reader as for the OLCI sensor. Basically we only add a yaml-file (meris_nc_sen3) and some minor changes to the OLCI-scripts

I will follow up with an extension of the TOA datasets in the upcoming weeks..

 - [x] Tests added 
 - [ ] Fully documented

### CODE REVIEW

**Comment by mraspaud** [comment]
@JohannesSMHI Great start, nice that so much of the Olci reader can be reused! Now we are just waiting for the tests :)

**Comment by JohannesSMHI** [comment]
@mraspaud Goodie! Should I create the tests as well, or will someone else do that?

**Comment by mraspaud** [comment]
I would say that you are probably the best person to do that :)

**Comment by JohannesSMHI** [comment]
@mraspaud would you consider this PR to be "Fully documented" ?

File: satpy/readers/meris_nc_sen3.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+References:
+    - :func:`xarray.open_dataset`
+"""
+
+import logging
+from functools import reduce
+
+import numpy as np
+
+from satpy.readers.olci_nc import (
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/readers/meris_nc_sen3.py:34]: @JohannesSMHI is this solved?

[the remark this answers is not in the document]
Comment (JohannesSMHI) [reply satpy/readers/meris_nc_sen3.py:34]: @mraspaud yep, that is solved.. but how about all the changes you did with the olci_l2. reader before the summer (eg. the BitFlags class) ? will the merge take care of that or do we need to adjust before we merge?

[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/readers/meris_nc_sen3.py:34]: Should be fine!

**Comment by gerritholl** [comment]
@JohannesSMHI This sadly seems to have some merge conflicts by now.  Any chance of looking into those?

**Comment by JohannesSMHI** [comment]
> 

@gerritholl, I´ll try to find time for this after my vaccation :)

**Comment by adybbroe** [comment]
@johannesocean @gerritholl and @mraspaud 
I just resolved some straight forward merge conflicts. Is this now getting ready to merge?

**Review by mraspaud — APPROVED** [review]
LGTM! Thanks @johannesocean for the PR and @adybbroe for finalizing!

**Comment by johannesocean** [comment]
@mraspaud @adybbroe Glad to hear this has been merged! :)

**Pull request merged**